src/wallet/wallet.cpp

```diff
@@ -1513,6 +1515,34 @@ bool CWallet::IsHDEnabled() const
     return !hdChain.seed_id.IsNull();
 }
 
+void CWallet::SetWalletFlag(uint64_t flags)
+{
+    LOCK(cs_wallet);
+    m_wallet_flags |= flags;
+    if (!WalletBatch(*database).WriteWalletFlags(m_wallet_flags))
+        throw std::runtime_error(std::string(__func__) + ": writing wallet flags failed");
+}
+
+bool CWallet::IsWalletFlagSet(uint64_t flag)
+{
+    return (m_wallet_flags & flag);
+}
+
+bool CWallet::SetWalletFlags(uint64_t overwriteFlags, bool memonly)
+{
+    LOCK(cs_wallet);
+    m_wallet_flags = overwriteFlags;
+    if ( (overwriteFlags >> 32 | g_known_wallet_flags >> 32) != (g_known_wallet_flags >> 32) ) {
```

Comment:
Code style is a bit funky here.

Reply:
Personally think this definitely needs parens to clarify priority, but maybe I'm just easily confused.

Reply:
Fixed